### PR TITLE
Etcd mTLS Operations Page

### PR DESCRIPTION
**Pull request by avestuk**
Add etcd operations page

Also re-order and re-organized Operations subitems to include an etcd folder.

### CODE REVIEW

**Comment by avestuk** [comment]
This is ready for review. The one thing that will need to be checked is the link to the deploy repository. I have set it to what I assume it'll be but if things change in that PR it may need updating.

File: _docs/operations/external-etcd/etcd-tls.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+StorageOS uses environment variables to specify the location of the necessary
+certificates and keys. The decision to use file paths instead of the actual
+certificate values was taken to prevent accidentally leaking the values. It is
+therefore required that Kubernetes secrets containing the certificates and keys
+are mounted as volumes into the StorageOS container and environment variables
+contain the file paths to the files from the secrets.
+
+## Setting up mTLS with etcd
+
+>In order to setup mTLS it is recomended that the CoreOS etcd operator is used
```
Comment (actionbuddha) [inline _docs/operations/external-etcd/etcd-tls.md]: Any reason we have this quoted? It reads slightly oddly.

File: _docs/operations/external-etcd/etcd-tls.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+>and the CoreOS guide for [Cluster
+>TLS](https://github.com/coreos/etcd-operator/blob/master/doc/user/cluster_tls.md)
+>is followed. You can find a worked example of setting up etcd and StorageOS
+>with TLS
+>[here](https://github.com/storageos/deploy/tree/master/k8s/deploy-storageos/etcd-tls)
+
+Once a TLS enabled etcd cluster is running a secret with the Certificate
```
Comment (actionbuddha) [inline _docs/operations/external-etcd/etcd-tls.md]: Needs a comma after 'running' I think.

File: _docs/operations/external-etcd/etcd-tls.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+Authority (CA) certificate, a client certificate and the client key needs to be
+created. The filenames used to create the secret become the key names in the
+secret. These filesnames need to be passed as part of the full file path to the
```
Comment (actionbuddha) [inline _docs/operations/external-etcd/etcd-tls.md]: s/filesnames/filenames/

File: _docs/operations/external-etcd/etcd-tls.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+etcd-client.crt:     798 bytes
+```
+> The secret should be created in the namespace that StorageOS will be
+> installed into
+
+Once the `etcd-client-tls` secret has been created it needs to be mounted into
+the pod. The path to each file mounted into the pod from the secret needs to be
+passed as an environment variable as in the table below.
+
+| Environement Variable | Value                         |
```
Comment (actionbuddha) [inline _docs/operations/external-etcd/etcd-tls.md]: Link to appropriate vars in the reference doc?

File: _docs/operations/external-etcd/etcd-tls.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+| :------------------   | :---------------------------- |
+| TLSEtcdCA             | /etc/pki/etcd-client.key      |
+| TLSEtcdClientCert     | /etc/pki/etcd-client-ca.crt   |
+| TLSEtcdClientKey      | /etc/pki/etcd-client.crt      |
+
+These environment variables can be passed in the daemonset manifest or as part
```
Comment (actionbuddha) [inline _docs/operations/external-etcd/etcd-tls.md]: s/daemonset/DaemonSet/

File: _docs/operations/external-etcd/etcd-tls.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+of a configMap. Below you can see an example of a configMap passing the
```
Comment (actionbuddha) [inline _docs/operations/external-etcd/etcd-tls.md]: /configMap/ConfigMap/

Comment (actionbuddha) [reply _docs/operations/external-etcd/etcd-tls.md]: (all occurences)

File: _docs/operations/external-etcd/etcd-tls.md (hunk 4 above)
Comment (actionbuddha) [reply _docs/operations/external-etcd/etcd-tls.md]: (do we have modifications to the reference doc?)

Comment (avestuk) [reply _docs/operations/external-etcd/etcd-tls.md]: Good point. I've added the envvars to the reference doc.

File: _docs/operations/external-etcd/etcd-tls.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+Once the `etcd-client-tls` secret has been created it needs to be mounted into
+the pod. The path to each file mounted into the pod from the secret needs to be
+passed as an [environment variable](/docs/reference/envvars) as in the table below.
+
+| Environement Variable | Value                         |
+| :------------------   | :---------------------------- |
+| TLSEtcdCA             | /etc/pki/etcd-client.key      |
+| TLSEtcdClientCert     | /etc/pki/etcd-client-ca.crt   |
+| TLSEtcdClientKey      | /etc/pki/etcd-client.crt      |
```
Comment (darkowlzz) [inline _docs/operations/external-etcd/etcd-tls.md]: Variable name and value order seems to be incorrect.

Comment (avestuk) [reply _docs/operations/external-etcd/etcd-tls.md]: Fixed.

File: _docs/reference/envvars.md [hunk 8]
```diff
@@ -45,3 +45,8 @@ variables. These can be injected in via any of the usual mechanisms such as
...
 | `PROBE_TIMEOUT_MS`        | 1.1.1             | The timeout to wait for an ack from a probed node before assuming it is unhealthy.  This should be set to 99-percentile of RTT (round-trip time) on your network.  Defaults to 3000ms. Added in `1.1.1` and deprecated in `1.1.2` See `PROBE_TIMEOUT`                                                         |
+| `TLSEtcdCA`               | 1.2.0             | The file path to the etcd server Certificate Authority certificate. The certificate must be mounted into the container using a ConfigMap or Secret. See [etcd TLS](/docs/operations/external-etcd/etcd-tls)                                                                                                                                                                                                                                           |
```
Comment (Arau) [inline _docs/reference/envvars.md]: Shouldn't this ENV variables be upper case?

Comment (actionbuddha) [reply _docs/reference/envvars.md]: Agreed. Also, I suggest we remove the configmap reference - we shouldn't be encouraging users to present keying material using anything other than a secret.

**Comment by Arau** [comment]
This PR has been merged from a different remote. Closing

**Pull request closed without merging**